src/List.ts

```diff
@@ -0,0 +1,106 @@
+import { Iterable, IterableIterator, ShimIterator } from 'dojo-shim/iterator';
+
+export interface List<T> extends Iterable<T> {
+	readonly size: number;
+
+	add(value: T): this;
+	clear(): void;
+	delete(idx: number): boolean;
+	entries(): IterableIterator<[ number, T ]>;
+	forEach(fn: (value: T, idx: number, list: this) => void, thisArg?: any): void;
+	has(idx: number): boolean;
+	includes(value: T): boolean;
+	indexOf(value: T): number;
+	join(separator?: string): string;
+	keys(): IterableIterator<number>;
+	lastIndexOf(value: T): number;
+	pop(): T | undefined;
+	push(value: T): void;
+	splice(start: number, deleteCount?: number, ...newItems: T[]): T[];
+	values(): IterableIterator<T>;
+}
+
+export default class ArrayList<T> implements List<T> {
+	private _items: T[];
+
+	[Symbol.iterator]() {
+		return this.values();
+	}
+
+	get size(): number {
+		return this._items.length;
+	}
+
+	constructor() {
```

Comment:
Shouldn't this be able to take an argument of an interable as a source, like the other iterables?  So you can do this:

```ts
new List([ 1, 2, 3 ]);
```